docs/guides/di-container.md

```diff
@@ -3,7 +3,9 @@ id: di-container
 title: Dependency injection container
 ---
 
-Dependency injection is a way of implementing inversion of control design pattern. It's used to handle the dependencies between multiple classes without the direct instantiation of one class in another - which causes tight coupling in the code, and makes it less testable and harder to maintain.
+Dependency injection is a way of implementing inversion of control design pattern. It's used to handle the dependencies between multiple classes without the direct instantiation of one class in another - which causes tight coupling in the code and makes it less testable and harder to maintain.
+
+In general, if you have something like this in your class `new SuperClass()` that is bad.
```

Comment:
I'd split it and reword it a bit:

---

In general, if you instantiate another class like this inside your class

```php
new SuperClass()
```

that should be avoided because it tightly couples two classes.

---

Even though it's mentioned in the paragraph above.